Makefile

```diff
@@ -76,3 +83,30 @@ branch-build: gen-content theme ## Builds a Git branch (for e.g. development bra
 		--gc \
 		--minify \
 		--enableGitInfo
+
+hugo:
+	git clone https://github.com/gohugoio/hugo.git -b v$(HUGO_VERSION)
+
+.PHONY: docker-support
+docker-support: hugo
+	docker build -t $(DEV_IMAGE_NAME) --build-arg HUGO_BUILD_TAGS=extended hugo/
+
+.PHONY: docker-image
+docker-image: docker-support
+	docker build -t $(PREVIEW_IMAGE_NAME) docker-support/
+
+.PHONY: docker-make-theme
+docker-make-theme:
+	docker run -v $(shell pwd):/site -it $(PREVIEW_IMAGE_NAME) make theme
+
+.PHONY: docker-make-serve
+docker-make-serve:
+	docker run -v $(shell pwd):/site -p 1313:1313 -it $(PREVIEW_IMAGE_NAME) make serve
```

Comment:
When you nest make within docker, you'll probably want to do
```Makefile
$(MAKE) \"MAKEFLAGS=$(MAKEFLAGS)\" serve
```
instead of
```Makefile
make serve
```

example: https://github.com/weaveworks/ignite/blob/2f81d05/Makefile#L100

Reply:
I think `$(MAKE)` is the wrong idea, I got

```
$ make docker-serve
docker run -v /Users/kingdonb/website:/site -p 1313:1313 -it fluxcd/website:hugo-support \
		/Library/Developer/CommandLineTools/usr/bin/make \"MAKEFLAGS=\" serve BIND_ADDRESS=0.0.0.0
docker: Error response from daemon: OCI runtime create failed: container_linux.go:367: starting container process caused: exec: "/Library/Developer/CommandLineTools/usr/bin/make": stat /Library/Developer/CommandLineTools/usr/bin/make: no such file or directory: unknown.
ERRO[0004] error waiting for container: context canceled
make: *** [docker-serve] Error 127
```

I don't think we need to pass in MAKEFLAGS or use the MAKE from the host, not sure why that would be needed, but I included the parts of what I think you suggested that didn't break as shown above.

I'm giving this one last test down each path before I'm ready to request a final review.